tools/SetupFlow/DevHome.SetupFlow/Models/CloningInformation.cs

```diff
@@ -89,7 +89,7 @@ public void SetIcon(ElementTheme theme)
             // Currently the only providers are Github and the generic provider.  The provider type
             // for the generic provider is git.
             // TODO: Remove when extensions have a GetIcon method.
-            if (ProviderName.Equals("github", StringComparison.OrdinalIgnoreCase))
+            if (RepositoryProvider.DisplayName.Equals("github", StringComparison.OrdinalIgnoreCase))

```

Comment:
If `RepositoryProvider` is OOP COM object, consider caching the display name (with a try/catch). That way all methods using the display name can get it from the in-proc copy and don't need to worry about try/catching COMException for every usage.

Reply:
Good idea.  Name is now cached.